src/openms/extern/tool_description_lib/src/tdl/convertToCWL.cpp

```diff
@@ -0,0 +1,503 @@
+                           },
+                           [&](DoubleValueList const &) {
+                               addInputArray(cwl::CWLType::double_);
+                           },
+                           [&](StringValueList const &) {
+                               if (child.tags.count("output")) {
+                                   if (child.tags.count("prefixed")) {
+                                       addOutputPrefixed(cwl::CWLType::File, /*.multipleFiles =*/true);
+                                   } else {
+                                       //! TODO not implemented
+                                   }
+                               } else if (child.tags.count("file")) {
+                                   addInputArray(cwl::CWLType::File);
+                               } else if (child.tags.count("directory")) {
+                                   addInputArray(cwl::CWLType::Directory);
+                               } else {
+                                   addInputArray(cwl::CWLType::string);
+                               }
+                           },
+                           [&](Node::Children const & recursive_children) {
+                               if (child.tags.count("basecommand")) {
+                                   baseCommandCB(child.name);
+
+                                   generateCWL<InputType, OutputType, deep - 1>(recursive_children,
+                                                                                doc,
+                                                                                inputCB,
+                                                                                outputCB,
+                                                                                baseCommandCB);
+                                   return;
+                               }
+
+                               auto inputs  = std::vector<cwl::CommandInputRecordField>{};
+                               auto outputs = std::vector<cwl::CommandOutputRecordField>{};
+
+                               generateCWL<cwl::CommandInputRecordField, cwl::CommandOutputRecordField, (deep - 1)>(
+                                   recursive_children,
+                                   doc,
+                                   [&](auto input) {
+                                       inputs.push_back(std::move(input));
+                                   },
+                                   [&](auto output) {
+                                       outputs.push_back(std::move(output));
+                                   },
+                                   baseCommandCB);
+
+                               auto inputType = cwl::CommandInputRecordSchema{};
+                               // auto outputType = cwl::CommandOutputRecordSchema{};
+
+                               inputType.fields = inputs;
+                               // outputType.fields = outputs; //!TODO Does this do anything?
```

Comment:
Are any of these TODO's things that need to be added for OpenMS?

Reply:
Good point, removed. All todos in TDL are not relevant for OpenMS.